### PR TITLE
fix(firestore-bigquery-export): fixed "Converting circular structure to JSON" error (issue #236).

### LINKED ISSUE #236: Error when mirroring data to BigQuery TypeError: Converting circular structure to JSON
### [REQUIRED] Step 2: Describe your configuration

- Extension name: **`firestore-bigquery-export`
- Extension version: **0.1.2**

### [REQUIRED] Step 3: Describe the problem

After running the initial setup and upon changing a field on a document in Firestore, the `ext-firestore-bigquery-export-fsexportbigquery` ran with the following error:

```
Error when mirroring data to BigQuery TypeError: Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
    at events.map (/srv/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/lib/bigquery/index.js:64:36)
    at Array.map (<anonymous>)
    at FirestoreBigQueryEventHistoryTracker.<anonymous> (/srv/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/lib/bigquery/index.js:56:33)
    at Generator.next (<anonymous>)
    at fulfilled (/srv/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/lib/bigquery/index.js:20:58)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:229:7)
```

This is the document for which the error occured:

Skärmavbild 2020-03-22 kl  22 07 55

### PR DESCRIPTION
**Pull request by russellwheatley**
Fixes #236

#### Inserting data into Firestore database:

Screenshot 2020-04-09 at 11 28 11

#### Result in BigQuery table:
Screenshot 2020-04-09 at 11 29 31

### CODE REVIEW

File: firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts [hunk 1]
```diff
@@ -58,20 +61,37 @@ export class FirestoreBigQueryEventHistoryTracker
...
 
+  serializeData(eventData: any) {
+    if (typeof eventData === "undefined") {
+      return undefined;
+    }
+
+    const data = traverse(eventData).map((property) => {
+      
+      if (property instanceof firebase.firestore.DocumentReference) {
+        return property.path;
```
Comment (MrAlek) [inline firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts]: 👌 Great job on finding the issue quickly @russellwheatley 🙏

**Comment by jhuleatt** [comment]
@IanWyszynski is going to document how to build & deploy without the need to publish the npm package. I believe the rough steps are:

1. in `firestore-bigquery-export/firestore-bigquery-change-tracker`:

`npm run build`

2. in `firestore-bigquery-export/package.json`:

change the `"@firebaseextensions/firestore-bigquery-change-tracker"` dependency to:

`"@firebaseextensions/firestore-bigquery-change-tracker": "file:./firestore-bigquery-change-tracker",`

3. in `firestore-bigquery-export/package.json`:

`npm run build`

and then deploy

**Comment by laurenzlong** [comment]
@russellwheatley Package has been published, you can test now.

File: firestore-bigquery-export/firestore-bigquery-change-tracker/package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   "main": "./lib/index.js",
   "scripts": {
     "build": "npm run clean && npm run compile",
-    "clean": "rimraf lib",
+    "clean": "rimraf lib/",
```
Comment (Salakar) [inline firestore-bigquery-export/firestore-bigquery-change-tracker/package.json:13]: Was this change necessary, should be working without this?

Comment (russellwheatley) [reply firestore-bigquery-export/firestore-bigquery-change-tracker/package.json:13]: Trying to make it more explicit it's a directory, but I'll change it back.

**Pull request merged**